File: Data Structures & Algorithms/is-anagram/submission-8.kt
class Solution {
    fun isAnagram(s: String, t: String): Boolean {
        val arr = IntArray(26)
        for(i in s){
            arr[i-'a']++
        }
        for(j in t){
            arr[j-'a']--
        }
        return arr.all{it == 0}
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-8.cpp
#include<algorithm>
using namespace std;

class Solution {
public:
    int minEatingSpeed(vector<int>& piles, int h) {
        int left = 1;
        int right = *max_element(piles.begin(), piles.end());

        return binarySearch(piles, left, right, h);
    }

    int binarySearch(vector<int>& piles, int left, int right, int h){
        if(left > right){
            return left;
        }
        int mid = (left + right) / 2 ;
        int hours = getCeil(piles, mid);
        if(hours <= h){
            return binarySearch(piles, left, mid - 1, h);
        }else{
            return binarySearch(piles, mid+1, right, h);
        }
    }

    int getCeil(vector<int>& arr, int h){
        int sum = 0;
        for(int i : arr){
            sum = sum + ((i + h - 1) / h);
        }
        return sum;
    }
};